portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,29 +58,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    BufferedReader body = request.getReader();
-    String content = body.readLine();
-    int endOfNameIndex = content.indexOf("*");
-    if (endOfNameIndex != 0 && endOfNameIndex != (content.length() -1 )) {
-      // Get name and text
-      String name = content.substring(0, endOfNameIndex++);
-      String text = content.substring(endOfNameIndex);
+    // Get name and text
+    String name = request.getParameter("name-input");
+    String text = request.getParameter("text-input");
 
-      long timestamp = System.currentTimeMillis();
+    long timestamp = System.currentTimeMillis();
 
-      Entity messageEntity = new Entity("Messages");
-      messageEntity.setProperty("name", name);
-      messageEntity.setProperty("text", text);
-      messageEntity.setProperty("timestamp", timestamp);
+    Entity messageEntity = new Entity("Messages");
+    messageEntity.setProperty("name", name);
```

Comment:
It probably doesn't matter for this case, but in general you'll want more precice variable names than that (and keys in storage are variable names!)

Ex: 
name => messageTitle
text => messageBody
timestamp => timestampMillis (the unit is implicit in the status quo, which is bad)